File: gestion_personnel_front/src/app/models/recording/recording.ts
import { Time } from "@angular/common";

export class Recording {
    id: number;
    date: string;
    dateStart: string;
    dateStop: string;
    hourStart: string;
    hourStop: string; 
    breakStart: string;
    breakStop: string;
    totalHours: string; 
    extraHours: string; 
    dueHours: string; 
    employeeId: number;
    dayTypeId: number;

    constructor(
        id: number,
        date: string,
        dateStart: string,
        dateStop: string,
        hourStart: string,
        hourStop: string,
        breakStart: string,
        breakStop: string,
        totalHours: string,
        extraHours: string,
        dueHours: string,
        employeeId: number,
        dayTypeId: number
    ) {
        this.id = id;
        this.date = date;
        this.dateStart = dateStart;
        this.dateStop = dateStop;
        this.hourStart = hourStart;
        this.hourStop = hourStop;
        this.breakStart = breakStart;
        this.breakStop = breakStop;
        this.totalHours = totalHours;
        this.extraHours = extraHours;
        this.dueHours = dueHours;
        this.employeeId = employeeId;
        this.dayTypeId = dayTypeId;
    }
}

File: gestion_personnel/src/main/java/com/cdaprojet/gestion_personnel/model/recording/Recording.java
package com.cdaprojet.gestion_personnel.model.recording;

import java.time.Duration;
import java.time.LocalDate;
import java.time.LocalTime;

import com.cdaprojet.gestion_personnel.model.dayType.DayType;
import com.cdaprojet.gestion_personnel.model.employeeModel.employee.Employee;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "recordings")
public class Recording {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private LocalDate date;
    private LocalDate dateStart;
    private LocalDate dateStop;
    private LocalTime hourStart;
    private LocalTime hourStop;
    private LocalTime breakStart;
    private LocalTime breakStop;
    private Duration extraHours;
    private Duration dueHours;
    private Duration totalHours;

    @ManyToOne
    @JoinColumn(name = "employee_id", nullable = true)
    private Employee employee;

    @ManyToOne
    @JoinColumn(name = "day_type_id", nullable = true)
    private DayType dayType;
}
